lib/bolt/apply_inventory.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'bolt/boltdir'
+require 'bolt/config'
+require 'bolt/error'
+
+module Bolt
+  class ApplyInventory
+    class InvalidFunctionCall < Bolt::Error
+      def initialize(function)
+        super("The function '#{function}' is not callable within an apply block",
+              'bolt.inventory/invalid-function-call')
+      end
+    end
+
+    attr_reader :config_hash
+
+    def initialize(config_hash = {})
+      @config_hash = config_hash
+      @targets = {}
+    end
+
+    def create_apply_target(target)
+      @targets[target.name] = target
+    end
+
+    def validate
+      @groups.validate
```

Comment:
I assume this method isn't necessary, given that its current implementation will error (this class has no instance variable called `@groups`).